nextest-metadata/src/test_list.rs

```diff
@@ -510,6 +516,26 @@ pub struct RustNonTestBinarySummary {
     pub path: Utf8PathBuf,
 }
 
+/// Serialized representation of the host and the target platform.
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct BuildPlatformsSummary {
+    /// The target platform, if specified.
+    pub target: PlatformSummary,
+
+    /// The libdir for the host platform.
+    ///
+    /// Empty if failed to discover.
+    #[serde(default)]
+    pub host_libdir: Option<Utf8PathBuf>,
```

Comment:
> also is independent justification to remove the "v2" stuff -- we're now clearly storing the host and target platforms separately, rather than smooshing them together.

IIUC, this new `platforms` field still makes both `target_platform` and `target_platforms` deprecated. Is that correct?

Reply:
Does this implicitly implies that `BuildPlatforms::host` should respect the new `HostPlatformSummary::platform`?

Reply:
In addition, what should be the interface of `from_summary` be like? `from_summary(summary: BuildPlatformsSummary) -> BuildPlatforms` or `from_summary(summary: BuildPlatformsSummary) -> Vec<BuildPlatforms>`? In the former case of `from_summary(summary: BuildPlatformsSummary) -> BuildPlatforms`, should we change `BuildPlatforms::target` from `Option<BuildPlatformsTarget>` to `Vec<BuildPlatformsTarget>`?

Reply:
> IIUC, this new platforms field still makes both target_platform and target_platforms deprecated. Is that correct?

Yeah.



> Does this implicitly imply that `BuildPlatforms::host` should respect the new `HostPlatformSummary::platform`?

Yep, while performing deserialization.



> In addition, what should be the interface of `from_summary` be like? `from_summary(summary: BuildPlatformsSummary) -> Result<BuildPlatforms>` or `from_summary(summary: BuildPlatformsSummary) -> Result<Vec<BuildPlatforms>>`?

The former.

> In the former case of `from_summary(summary: BuildPlatformsSummary) -> Result<BuildPlatforms>`, should we change `BuildPlatforms::target` from `Option<BuildPlatformsTarget>` to `Vec<BuildPlatformsTarget>`?

Just error out if there's more than one.

Reply:
> Just error out if there's more than one.

Which error should I return? `RustBuildMetaParseError::PlatformDeserializeError` or `RustBuildMetaParseError::UnknownHostPlatform` or should I create another enum variant? In case of using existing enum variant, how do I generate the source `target_spec::Error` or should I change the variant to be another enum which includes a `MultipleTargets` variant?

Reply:
Another variant on RustBuildMetaParseError is fine.

Reply:
Done.

* Introduce new types `HostPlatformSummary` and `TargetPlatformSummary`
* Deserialize `BuildPlatforms::host` from `HostPlatformSummary::platform`
* Introduce a new  `RustBuildMetaParseError` variant, which is returned if the argument of `from_summary(summary: BuildPlatformsSummary) -> Result<BuildPlatforms>` has more than one entries in the `targets` field.